src/xinterpreter.cpp

```diff
@@ -131,6 +131,42 @@ namespace xeus
         }
     }
 
+    nl::json xinterpreter::publish_execution_output(int execution_count,
+                                                const std::string& status,
+                                                const std::string& ename,
+                                                const std::string& evalue,
+                                                const std::string& data_type,
+                                                const std::vector<std::string>& data_content,
+                                                const std::vector<std::string>& trace_back)
+    {
+        if (m_publisher)
+        {
+            nl::json content;
+            content["status"] = status;
+            content["ename"] = ename;
+            content["evalue"] = evalue;
+            content["traceback"] = nl::json::array();
+
+            if (content["status"] == "error")
+            {
+                content["traceback"] = trace_back;
+            }
+            else if (content["status"] == "execute_reply")
+            {
+                content["execution_count"] = execution_count;
+                content["payload"] = nl::json::array();
+                content["user_expressions"] = nl::json::object();
+            }
+            else if (content["status"] == "abort")
+            {
+                return content;
+            }
+
+            content[data_type] = data_content;
+            return content;
```

Comment:
Every time the user calls this method they will also want to publish the result. I wonder if we should add `publish_execution_result` here or let them add it themselves.